modules/pel/peltool/peltool.py

```diff
@@ -586,13 +586,25 @@ def main():
     if args.hidden:
         config.hidden = True
 
+    if args.file:
+        try:
+            with open(args.file, 'rb') as fd:
+                data = fd.read()
+                stream = DataStream(data, byte_order='big', is_signed=False)
+                _, json_string = parsePEL(stream, config, True)
+                print(json_string)
+        except Exception as e:
+            print(f"Exception: No PEL parsed for {args.file}: {e}")
+        if args.clean:
+            os.remove(args.file)
+        sys.exit(0)
+
     if not inBMC:
-        if not args.file:
-            if not args.path:
-                sys.exit("Outside the BMC environment, please provide the path to the PELs using the -p option.")
-            if not os.path.isdir(args.path):
-                sys.exit(f"{args.path} is not a valid directory")
-            PELsPath = args.path
+        if not args.path:
```

Comment:
Suggestion: A better approach might be to use [Action](https://docs.python.org/3/library/argparse.html#action) to define a custom action for the argument, similar to what you've done here.

Reply:
For -f/--file I feel 'store' action seems fit. However, this is the default action setting, therefore didn't specified action for this option. 
Please let me know if you feel other action seems better fit for this option.